analyzers/src/SonarAnalyzer.Common/Rules/ExpressionComplexityBase.cs

```diff
@@ -22,8 +22,8 @@
 
 namespace SonarAnalyzer.Rules
 {
-    public abstract class ExpressionComplexityBase<TExpression, TSyntaxKind> : ParameterLoadingDiagnosticAnalyzer where TSyntaxKind : struct
-        where TExpression : SyntaxNode
+    public abstract class ExpressionComplexityBase<TSyntaxKind> : ParameterLoadingDiagnosticAnalyzer
+        where TSyntaxKind : struct, Enum
```

Comment:
you've added `Enum` - did you add a test for this?

Reply:
> if I remove Enum from the code, I expect a test should fail, right?

If I remove it nothing happens. It is there to make sure that the type argument passed here is an enum: 
`public sealed class ExpressionComplexity : ExpressionComplexityBase<SyntaxKind>`
without the Enum constraint you could do this:
`public sealed class ExpressionComplexity : ExpressionComplexityBase<int>`